src/platform/posix/tcp_ip_bundle.c

```diff
@@ -83,22 +84,25 @@ lf_ret_t TcpIpBundle_send(TcpIpBundle *self, PortMessage *message) {
     socket = self->fd;
   }
 
-  // turing write buffer into pb_ostream buffer
-  pb_ostream_t stream = pb_ostream_from_buffer(self->write_buffer, TCP_BUNDLE_BUFFERSIZE);
-
   // serializing protobuf into buffer
-  int status = pb_encode(&stream, PortMessage_fields, message);
+  int message_size = encode_protobuf(message, self->write_buffer, TCP_BUNDLE_BUFFERSIZE);
 
-  if (status < 0) {
+  if (message_size < 0) {
     return LF_ERR;
   }
 
   // sending serialized data to client
-  ssize_t bytes_written = write(socket, self->write_buffer, stream.bytes_written);
+  ssize_t bytes_written = 0;
+  int timeout = TCP_IP_TIMEOUT;
+
+  while (bytes_written < message_size && timeout > 0) {
+    bytes_written += write(socket, self->write_buffer + bytes_written, message_size - bytes_written);
+    timeout--;
+  }
```

Comment:
This also seems to bound the size of the messages we can write? Should the timeout really be decremented unless `write` returns `-1`?